### PR TITLE
Move API Reference

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Relocate https://github.com/mattermost/mattermost-api-reference into the monorepo.

#### Ticket Link
None.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by lieut-data** [comment]
@cwarnermm & @toninis, this is a straightforward relocation of https://github.com/mattermost/mattermost-api-reference into the monorepo, with the hopes that it becomes simpler to maintain going forward.

One unsolved problem: currently the https://git.internal.mattermost.com/mattermost/ci-only/mattermost-api-reference repo is responsible for deploying the resulting artifacts to S3 on push to master. In principle, this will start working from the https://git.internal.mattermost.com/mattermost/ci-only/mattermost-server repo /if/ we have the appropriate secrets setup. Or should we evaluate moving this all to GitHub actions as part of this effort instead?

**Review by cwarnermm — APPROVED** [review]
Thanks, @lieut-data! I'll defer to @toninis for the outstanding challenge ahead.

File: .github/workflows/api.yml
Comment (toninis) [inline .github/workflows/api.yml:26]: You can introduce cache here: 
```suggestion
      with:
        node-version: '16.2.0'
        cache: 'npm'
        cache-dependency-path: package-lock.json
```

Comment (lieut-data) [reply .github/workflows/api.yml:26]: Nice!

**Comment by lieut-data** [comment]
> Hello @lieut-data we can easily use our new Unified CI framework in order to add this process on our private repo/runners and ditch Gitlab . I can quickly create an automation upon master push that will trigger the API pipeline privately . This is a fine example to start using the Unified CI framework by your self also . I can make the start and we can share the knowledge together to see how this works.

Yes, I would love to learn how to do this, @toninis! If you can kickstart me, I'm happy to keep going until I get stuck.

> As I understand:
> 
>     * This is going to be triggered on master push correct ?

Yes, that's correct.

**Comment by toninis** [comment]
@lieut-data Check https://github.com/mattermost/delivery-platform/pull/5 . This should handle the release workflow of the API reference . If you check the conditions it will run after successful master execution of the API workflow. 

It needs to be merged before this one is merged. If you need any clarification ping me.

File: .github/workflows/api.yml
Comment (phoinixgrr) [inline .github/workflows/api.yml:25]: Could we avoid hardcoding versions in the CI process unless there is a valid reason for it, as mentioned in this discussion: https://github.com/mattermost/delivery-platform/pull/5#discussion_r1236750888

File: .github/workflows/api.yml
Comment (phoinixgrr) [inline .github/workflows/api.yml:21]: Lets SHA PIN this action used here

File: .github/workflows/api.yml
Comment (phoinixgrr) [inline .github/workflows/api.yml:23]: Lets SHA PIN this action used here

File: .github/workflows/api.yml
Comment (phoinixgrr) [inline .github/workflows/api.yml:30]: If not using multiple lines, let's make it this single line.

File: .github/workflows/api.yml
Comment (phoinixgrr) [inline .github/workflows/api.yml:32]: Lets SHA PIN this action used here

File: api/.gitpod.yml
Comment (phoinixgrr) [inline api/.gitpod.yml:8]: If not using multiple lines, let's make it this single line.

File: api/.gitpod.yml
Comment (phoinixgrr) [inline api/.gitpod.yml:10]: If not using multiple lines, let's make it this single line.

**Review by phoinixgrr — APPROVED** [review]

File: api/.gitlab-ci.yml
Comment (toninis) [inline api/.gitlab-ci.yml:1]: We can remove this 🚀 in favour of https://github.com/mattermost/delivery-platform/pull/5

**Pull request merged**
